action.yml

```diff
@@ -172,6 +176,13 @@ runs:
         role-session-name: "atmos-terraform-plan-gitops"
         mask-aws-account-id: "no"
 
+    - name: Configure Google Credentials
+      if: ${{ steps.config.outputs.backend == 'google' }}
+      uses: google-github-actions/auth@v2
+      with:
+        workload_identity_provider: ${{ steps.config.outputs.google-workload-identity-provider }}
+        service_account: ${{ steps.config.outputs.google-service-account }}
+
```

Comment:
This is a bit confusing part. Let's discuss.
The current atmos gitops pattern assume there are 2 types of AWS creds 
```
integrations:
  github:
    gitops:
....
      artifact-storage:
....
        role: arn:aws:iam::xxxxxxxxxxxx:role/cptest-core-ue2-auto-gitops-gha
      role:
        plan: arn:aws:iam::yyyyyyyyyyyy:role/cptest-core-gbl-identity-gitops
        apply: arn:aws:iam::yyyyyyyyyyyy:role/cptest-core-gbl-identity-gitops
....
```

1. IAM role `integration.github.gitops.artifact-storage.role` that used to read/write plan file to storage
2. IAM role `integration.github.gitops.role.plan` / `integration.github.gitops.role.apply` that used to run terraform plan / apply.

The second ones have wider permissions.
Really, eventually, we'd like to deprecate that part of the config.
Allow external AWS auth actions because we can not support all levels of AWS auth.

I had a feeling that in the case of Google Cloud, you mixed those two types of auth.
I think we do not need the step.

Reply:
Moved auth step to GH worklfow in our own project. As you were saying it is not needed here.